Druid/DRBytecodeGenerator.class.st

```diff
@@ -251,11 +258,20 @@ DRBytecodeGenerator >> visitReturn: aDRReturn [
 
 { #category : 'ir-to-target' }
 DRBytecodeGenerator >> visitStoreInstVar: aDRStoreInstanceVariable [ 
-
+	
 	aDRStoreInstanceVariable operand1 acceptVisitor: self.
-	builder pushLiteral: aDRStoreInstanceVariable operand2 value.
-	aDRStoreInstanceVariable operand3 acceptVisitor: self.
-	builder send: #instVarAt:put:
+	aDRStoreInstanceVariable operand1 class = DRLoadTemporaryVariable ifTrue: [ 
+		builder pushLiteral: aDRStoreInstanceVariable operand2 value.
+		aDRStoreInstanceVariable operand3 acceptVisitor: self.
+		builder send: #instVarAt:put: .
+		builder popTop.
+	]
+	ifFalse: [ 
+		aDRStoreInstanceVariable operand3 acceptVisitor: self.
+		builder storeInstVar: aDRStoreInstanceVariable operand2 value.
+		builder popTop.
+	].
```

Comment:
I suggest to write these methods in the form:

```st
CONDITION ifTrue: [
   ... TRUE CODE ...
  ^ self
]
... FALSE CODE ...
```

_Generally_, a simple ifTrue: is easier to read than an `ifTrue:ifFalse:`